com.unity.netcode.gameobjects/Components/HalfVector3.cs

```diff
@@ -0,0 +1,141 @@
+using System.Runtime.CompilerServices;
+using UnityEngine;
+
+namespace Unity.Netcode.Components
+{
+    /// <summary>
+    /// Structure that defines which axis of a <see cref="HalfVector3"/> will
+    /// be serialized.
+    /// </summary>
+    public struct HalfVector3AxisToSynchronize
+    {
+        /// <summary>
+        /// When enabled, serialize the X axis
+        /// </summary>
+        public bool X;
+
+        /// <summary>
+        /// When enabled, serialize the Y axis
+        /// </summary>
+        public bool Y;
+
+        /// <summary>
+        /// When enabled, serialize the Z axis
+        /// </summary>
+        public bool Z;
+
+        /// <summary>
+        /// Constructor to preinitialize the x, y, and z values
+        /// </summary>
+        /// <param name="x">when <see cref="true"/> the x-axis will be serialized</param>
+        /// <param name="y">when <see cref="true"/> the y-axis will be serialized</param>
+        /// <param name="z">when <see cref="true"/> the z-axis will be serialized</param>
+        public HalfVector3AxisToSynchronize(bool x = true, bool y = true, bool z = true)
+        {
+            X = x;
+            Y = y;
+            Z = z;
+        }
+    }
+
+    /// <summary>
+    /// Half float precision <see cref="Vector3"/>
+    /// </summary>
+    public struct HalfVector3 : INetworkSerializable
+    {
+        /// <summary>
+        /// The half float precision value of the x-axis as a <see cref="ushort"/>
+        /// </summary>
+        public ushort X;
+        /// <summary>
+        /// The half float precision value of the y-axis as a <see cref="ushort"/>
+        /// </summary>
+        public ushort Y;
+        /// <summary>
+        /// The half float precision value of the z-axis as a <see cref="ushort"/>
+        /// </summary>
+        public ushort Z;
+
+        private HalfVector3AxisToSynchronize m_HalfVector3AxisToSynchronize;
+
+        /// <summary>
+        /// The serialization implementation of <see cref="INetworkSerializable"/>
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="serializer"></param>
+        public void NetworkSerialize<T>(BufferSerializer<T> serializer) where T : IReaderWriter
+        {
+            if (m_HalfVector3AxisToSynchronize.X)
```

Comment:
That is a good question.  Since the [axis synchronized are defined by NetworkTransform](https://github.com/Unity-Technologies/com.unity.netcode.gameobjects/blob/0d9709acfa2cb8e3348d0b996b3c18adf7475daa/com.unity.netcode.gameobjects/Components/NetworkTransform.cs#L613) during serialization (and my mind was thinking about bandwidth optimizations at the time) it seemed like a waste of additional bandwidth to synchronize which axis were being synchronized each time it was serialized.  
Definitely open to discussing if we  want to somehow make it possible to synchronize this during serialization, but currently with half float precision we get roughly 25 bytes per full transform update (21 bytes with Quaternion compression) per instance.  For the most part those numbers are about as packed as they can get (give or take a couple of bits), adding any additional information (i.e. which axis to synchronize) would increase those numbers by 1 byte per instance.
However, since I made the struct public a user might want to use that for some other purpose than what it is being used for... so we could add an additional bool, via public get/set or constructor, that defaults to false and would serialize m_HalfVector3AxisToSynchronize first if set to true.  

Thoughts?

Reply:
Yeah I was thinking its just a visibility issue, its perfectly fine to assume they will be the same on both ends as long as we provide some method to detect it isn't, even if that's only in debug builds or with error checking enabled.

Reply:
Since authority dictates which axis are updated,  all to-be-synchronized axis are included, whether each to-be-synchronized axis has a delta or not, on the authority side when any of the axis have a delta.  On the non-authoritative side, the HasPositionX, HasPositionY, HasPositionZ all define which axis to expect and the HalfVector3 or HalfVector3DeltaPosition initialize with that specific HalfVector3AxisToSynchronize setting prior to serialization (reading).